common/metrics/tags.go

```diff
@@ -135,6 +137,16 @@ func DecisionTypeTag(value string) Tag {
 	return metricWithUnknown(decisionType, value)
 }
 
+// ShardscannerScanResult returns a new shardscanner scan result type tag.
+func ShardscannerScanResult(value string) Tag {
```

Comment:
small thing but perhaps capitalize scanner (and all other relevant areas within PR):
ShardScannerScanResult